server/etcdserver/server.go

```diff
@@ -2136,10 +2136,10 @@ func (s *EtcdServer) snapshot(snapi uint64, confState raftpb.ConfState) {
 			"saved snapshot",
 			zap.Uint64("snapshot-index", snap.Metadata.Index),
 		)
-		s.updateStorageSchema.Do(func() {
-			// Update storage schema after snapshot as fields introduced in v3.5 should be set.
-			// Remove in v3.7
-			err := schema.UpdateStorageSchema(s.lg, s.be.BatchTx())
+		s.migrateStorageSchema.Do(func() {
+			// Update storage schema after snapshot as setting it requires all fields introduced in v3.5 to be set. Those field can only be set after snapshot.
+			// TODO: Remove in v3.7 as we should be able to depend on running Migrate during etcdserver boostrap.
+			err := schema.Migrate(s.lg, s.be.BatchTx(), schema.V3_6)
```

Comment:
Shell we assert here based on:
  https://github.com/etcd-io/etcd/blob/main/api/version/version.go
  
  As this code overlooked in 3.7 upgrade process might be dangerous (with hardcoded to 3.6 migration)

Reply:
Done